web/src/lib/components/shared-components/theme-button.svelte

```diff
@@ -1,21 +1,46 @@
 <script lang="ts">
+  import { onMount } from 'svelte';
   import { browser } from '$app/environment';
   import { colorTheme } from '$lib/stores/preferences.store';
   import IconButton from '../elements/buttons/icon-button.svelte';
 
+  import { appearanceSettings } from '../../stores/preferences.store';
+
   const toggleTheme = () => {
     $colorTheme = $colorTheme === 'dark' ? 'light' : 'dark';
   };
 
-  $: {
+  function checkSystemTheme() {
+    if($appearanceSettings.autoTheme) {
+      const darkModeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+      $colorTheme = darkModeMediaQuery.matches ? 'dark' : 'light';
+      changeTheme()
+    }
+  };
+
+  function changeTheme() {
     if (browser) {
       if ($colorTheme === 'light') {
         document.documentElement.classList.remove('dark');
       } else {
         document.documentElement.classList.add('dark');
       }
     }
+  };
+
+  $: {
+    const theme = $colorTheme;
+    changeTheme();
+  }
+
+  $: {
+    const autoTheme = $appearanceSettings.autoTheme;
+    checkSystemTheme();
   }
+
+  onMount(() => {
+      window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', checkSystemTheme);
```

Comment:
This only adds the event listener if the toggle is shown which seems wrong. Can you move this code to the layout.svelte instead?

Also can you use svelte:window component to add the listener instead? Also you should not be manually updating the class list, that is not very svelte like.

Reply:
you mean the "user-page-layout.svelte"?
For the second point, yes, I'm gonna change it. Sorry, but I'm quite new to svelte. I wanna learn it to contribute more

Reply:
In +layout.svelte code at the top of the file will run once on page load. That seems like a good place to put the initial theme stuff. Also, I think you can use svelte:window in this component as well. We only need to register the event once, not every time the theme switcher is mounted.

Reply:
I think we should maybe able to also bind the document class by setting up something like this, also in the layout component.

```svelte
<svelte:document class:dark={isDark} />
```